torch/csrc/jit/symbolic_variable.h

```diff
@@ -37,9 +37,24 @@ struct SymbolicVariable {
       }
       return out;
   }
+  static bool isScalarValue(at::Scalar s, double v) {
+    if(s.isFloatingPoint()) {
+      return v == s.toDouble();
+    } else {
+      return v == (double) s.toLong();
+    }
+  }
   SymbolicVariable operator*(SymbolicVariable rhs) const {
     return create(kmul, {*this, rhs})[0].typeLike(*this);
   }
+  SymbolicVariable operator*(at::Scalar rhs) const {
+    if(isScalarValue(rhs, 1))
+      return *this;
```

Comment:
It is less work to put it here, and pretty common  (look at LLVM's builder) to do these cleanups as we see them. Otherwise you wait an unknown amount of time before then cleanup happens, and some other pass may have obscured an optimization that was seen before.